src/ngrest/render/RenderCrud.php

```diff
@@ -365,6 +365,102 @@ public function getCompositionKeysForButtonActions($modelPrefix)
+     * 
+     * Conditions needs to be defined in the model's ngRestScopes() along with fields definition
+     * indexed by buttonCondition and provide as a string or a [field=>value] array.
+     * ```php
+     * public function ngRestScopes()
+     * {
+     *     return [
+     *       ['update', ['field1', 'field2', 'field3'], ['buttonCondition' => ['{created_by}' => Yii::$app->adminuser->id]] ]
+     *     ]
+     * } 
+     * ```
+     * 
+     * 
+     * Conditions may be defined also in the model ngRestConfigOptions() along with other
+     * options
+     * Exemple :
+     *
+     * ```php
+     *    public function ngRestConfigOptions()
+     *    {
+     *        return [
+     *            // ...
+     *            'buttonsCondition' => [
+     *                [ ['update', 'delete'], '{created_by}=='. \Yii::$app->adminuser->id  ],
+     *            ],
+     *        ];
+     *    }
+     * ```
+     * 
+     * This will add an ng-Show = "item.created_by==1" for instance if the logged user is the admin.
+     *
+     *
+     * @param string $scope The scope aka button context like 'create' or 'update' used in the used
+     * in the model ngRestConfigOptions() definition
+     * @return string Returns the condition with replaced field context like `item.create_id== 1` or  'true'
+     * @throws InvalidConfigException
+     * @since 1.2.0
+     */
+    public function getListButtonNgShowCondition($scope)
+    {
+
+               
+        $buttonConditionConfigOption = $this->config->getOption('buttonCondition');
+
+        // return true of no condition is defined
+        if (empty($buttonConditionConfigOption)){
+            return 'true';
+        }
+        // throw exception if configuration is wrong
+        else if ( !is_array($buttonConditionConfigOption) ) {
+            throw new InvalidConfigException("Invalid buttonsCondition ngRestConfigOptions definition.");
+        }
+
+
+
+        $buttonCondition = '';
+        foreach ($buttonConditionConfigOption as $arrayConfig) {
+            if ( count ($arrayConfig) < 2) {
+                 throw new InvalidConfigException("Invalid buttonsCondition ngRestConfigOptions definition. buttonsCondition must be an array with two elements similar to ngRestScopes. Ex. `[ ['update', 'delete'], '{fieldname} == 1'´]`");
+            }
+
+            // take the first entry of the array config as the config scope
+            // and check whether it applys to the given scppe
+            $configScope = array_shift($arrayConfig);
+            if ($configScope == $scope || (is_array($configScope) && in_array($scope, $configScope))) {
+                $buttonCondition = array_shift($arrayConfig);
+                break;
+            }
+        }
+
+
+        if ( !$buttonCondition) {
+            return 'true';
+        }
+
+        // prepend context to $field if in format '{fieldname}'
+        preg_match_all('/{(.*?)}/', $buttonCondition, $matches, PREG_SET_ORDER);
```

Comment:
we do have already a function for this, maybe we can share this informations or create a helper for both? or a STATIC function within the RenderCrud or Config class.

see here https://github.com/luyadev/luya-module-admin/blob/master/src/ngrest/base/Plugin.php#L378-L452

Reply:
Indeed wanted to do so  but It was not a static function so I duplicated the code for testing but I agree it could be better to have a single static function for  both use.

Reply:
Had implemeted the code reuse in Config class in just pushed to this pulrequest.